### ISSUE TITLE
Request for .stldr File Generation for STM32H745ZI-Q with W25Q128JV Flash Memory

### ISSUE THREAD

**Issue by kzorer** [cturvey/stm32extldr#10]
Hi,
I am trying to generate a .stldr file for the STM32H745ZI-Q device, paired with the external flash memory model W25Q128JV. Below is the pin configuration I am using for QUADSPI:

- PB2 → QUADSPI_CLK
- PC11 → QUADSPI_NCS
- PE7 → QUADSPI_IO0
- PE8 → QUADSPI_IO1
- PE9 → QUADSPI_IO2
- PE10 → QUADSPI_IO3

I am encountering issues during this process and would appreciate guidance on what might be going wrong. Additionally, if possible, could you provide a .stldr file for the above configuration?

Thank you for your support!

**Comment by cturvey** [comment]
Yeah, will take a gander this evening.

PB2:AF9 CLK
PC11:AF9 BK2 NCS
PE7:AF10 BK2 IO0
PE8:AF10 BK2 IO1
PE9:AF10 BK2 IO2
PE10:AF10 BK2 IO3

My hot-take would be an issue with the pin settings, or the second bank select

**Comment by cturvey** [comment]
Try this
https://github.com/cturvey/stm32extldr/blob/main/h7_w25q128/CLIVEONE-W25Q128_STM32H7XX-PB2-PC11-PE7-PE8-PE9-PE10.stldr

**Comment by kzorer** [comment]
I uploaded directly into "..\STM32Cube\STM32CubeProgrammer\bin\ExternalLoader" file but it does not work. It gives an error such that: "Error: Data read failed"

**Comment by cturvey** [comment]
Ok, will check over later
Has this design been tested/validated with your own BSP code? Bringing up the QSPI on the application side?

**Comment by cturvey** [comment]
Perhaps can you screen shot this in STM32 Cube Programmer, with Verbose Level 3, to provide additional context.

**Comment by kzorer** [comment]
Yes I have already been tested QSPI peripheral. I'm reading and writing to the flash properly. Here is the screen shot from STM32CubeProgrammer

![image](https://github.com/user-attachments/assets/b855f414-f87f-4871-ac00-6abddedc0852)

**Comment by cturvey** [comment]
Ok, so failing in Init()
What HSE source / frequency?
SMPS power supply?
What JEDEC ID does the Winbond report? Command 0x9F

**Comment by kzorer** [comment]
Hi again,
My HSE source frequency is 8MHz.
SMPS power supply --> 12V/1A
JEDEC ID --> 7018h that means W25Q128JV-IM*/JM* model reported.

**Comment by cturvey** [comment]
Ok, updated loader, was looking for EF 60/40 xx patterns, not EF 70 xx
Please try the new build
Do you have USART available for diagnostics? Which USART and Pins?

**Comment by kzorer** [comment]
I have observed some behavior related to starting addresses during connection. When I initially connect using the starting address 0x08000000 and then switch to 0x90000000, the connection fails. However, if I connect directly to 0x90000000 from the start, it works without issues. Is this behavior normal, and what could be the underlying reason for this sequence dependency?

Additionally, could you explain how you generate the loader file? You previously asked me some questions related to this, but I am unsure how they are connected to the loader file generation process.

**Comment by cturvey** [comment]
I have a relatively flat approach using the HAL libraries to create a "DLL" exporting the loader functions and without using interrupts.
Use make + GNU/GCC to compile, and process the STLDR file
No startup.s or main.c, so no Vectors, Reset_Handler, no main()
Loader_Src.c / Loader_Src.h containing the loaders entry points, QSPI code, pin definitions and part details
Dev_Inf.c / Dev_Inf.h containing the headering details
ExternalLoader.ld for the linker template, see example https://github.com/cturvey/stm32extldr/blob/main/ExternalLoader.ld
I can optionally use a UART to output debug/telemetry to understand issues and flow dynamics
